### PR TITLE
Add AnnotatedConverterFactory sample

### PR DESCRIPTION
**Pull request by NightlyNexus**
Stole the idea and most of the code from one of Jake's recent talks.

Not sure what you had in mind here, or if you want it in the samples at all, but here it is in case it's helpful to anyone else.

### CODE REVIEW

File: samples/src/main/java/com/example/retrofit/AnnotatedConverters.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
+package com.example.retrofit;
```
Comment (JakeWharton) [inline samples/src/main/java/com/example/retrofit/AnnotatedConverters.java]: license

File: samples/src/main/java/com/example/retrofit/AnnotatedConverters.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          return factory.responseBodyConverter(type, annotations, retrofit);
+        }
+      }
+      return null;
+    }
+
+    @Override public Converter<?, RequestBody> requestBodyConverter(Type type,
+        Annotation[] parameterAnnotations, Annotation[] methodAnnotations, Retrofit retrofit) {
+      for (Annotation annotation : parameterAnnotations) {
+        Converter.Factory factory = factories.get(annotation.annotationType());
```
Comment (NightlyNexus) [inline samples/src/main/java/com/example/retrofit/AnnotatedConverters.java]: Oops, I would have done a patch commit if I had seen you had seen this first.
I fixed this just now; I had `annotation.getClass()` before.

File: samples/src/main/java/com/example/retrofit/AnnotatedConverters.java [hunk 3]
```diff
@@ -1,3 +1,18 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
```
Comment (JakeWharton) [inline samples/src/main/java/com/example/retrofit/AnnotatedConverters.java]: 2017

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**